react/features/display-name/components/AbstractDisplayNamePrompt.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+
+import { Component } from 'react';
+
+import { updateSettings } from '../../base/settings';
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link AbstractDisplayNamePrompt}.
+ */
+export type Props = {
+
+    /**
+     * Invoked to update the local participant's display name.
+     */
+    dispatch: Dispatch<*>,
+
+    /**
+     * Function to be invoked after a successful display name change.
+     */
+    onPostSubmit: ?Function,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function
+};
+
+/**
+ * Implements an abstract class for {@code DisplayNamePrompt}.
+ */
+export default class AbstractDisplayNamePrompt<S: *>
+    extends Component<Props, S> {
+    /**
+     * Instantiates a new component.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._onSetDisplayName = this._onSetDisplayName.bind(this);
+    }
+
+    _onSetDisplayName: string => boolean;
+
+    /**
+     * Dispatches an action to update the local participant's display name. A
+     * name must be entered for the action to dispatch.
+     *
+     * @param {string} displayName - The display name to save.
+     * @returns {boolean}
```

Comment:
Why does this return boolean? For some dialog interaction? If so, can you please mention it there?

Reply:
Sure, added a comment there